web-registry/src/lib/tracker/types.ts

```diff
@@ -0,0 +1,54 @@
+export type Track = <IEventName extends string, IPayload = void>(
+  eventName: IEventName,
+  payload?: IPayload,
+) => Promise<any>;
+
+export interface LoginEvent {
+  date: string;
+  account: string;
+}
+
+export interface Buy1Event {
+  url: string;
+  cardType?: string;
+  buttonLocation: string;
+  projectName?: string | null;
+  projectId?: string | null;
+  creditClassName?: string;
+  creditClassId?: string | null;
+}
+
+export interface Buy2Event {
+  url: string;
+  price?: string;
+  batchDenom?: string;
+  projectName?: string | null;
+  projectId?: string | null;
+  creditClassId?: string;
```

Comment:
there are some call sites for the `track` function where the credit class name is not readily available, and where i find it difficult to figure out how to efficiently pass it to. open to suggestions on how to improve that for these cases where i did not include it. but since we do have `projectId` and `creditClassId` available in almost all places, i almost wonder if we can just adjust the expectation in our analytics so that we don't include the credit class name. it seems to me like the ID is the more important field anyways, from an analytics perspective. it's a simpler key to aggregate on, etc.

my concern on the technical side was that i didn't want to add an api call to lookup the name prior to sending the tracking event, and introduce a performance issue.

Reply:
yeah I agree just including the (on chain) ids would be sufficient
any thoughts @erikalogie ?